torch/nn/modules/conv.py

```diff
@@ -1169,11 +1169,23 @@ def initialize_parameters(self, input) -> None:  # type: ignore[override]
             else:
                 self.weight.materialize((
                     self.out_channels, self.in_channels // self.groups, *self.kernel_size))
+            self.weight = torch.nn.Parameter(self.weight.to(memory_format=self._get_in_memory_format(input)))
             if self.bias is not None:
                 assert isinstance(self.bias, UninitializedParameter)
                 self.bias.materialize((self.out_channels,))
             self.reset_parameters()
 
+    # Function to extract memory format from first input.
+    def _get_in_memory_format(self, input: Tensor) -> torch.memory_format:
+        memory_format = torch.contiguous_format
+        if self._get_num_spatial_dims() == 2:
+            if input.is_contiguous(memory_format=torch.channels_last):
+                memory_format = torch.channels_last
```

Comment:
this is too aggressive and will label all NC11 inputs as channels_last, you probably need to expose `suggest_memory_format` from c++. Primtorch has similar utility in python that also could be used.